editor/components/inserter-with-shortcuts/index.js

```diff
@@ -0,0 +1,76 @@
+/**
+ * External dependencies
+ */
+import { connect } from 'react-redux';
+import { filter, isEmpty } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { BlockIcon, createBlock, getDefaultBlockName } from '@wordpress/blocks';
+import { compose } from '@wordpress/element';
+import { IconButton, withContext } from '@wordpress/components';
+import { __, sprintf } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import './style.scss';
+import Inserter from '../inserter';
+import { getFrequentInserterItems } from '../../store/selectors';
+import { replaceBlocks } from '../../store/actions';
+
+function InserterWithShortcuts( { items, isLocked, onToggle, onInsert } ) {
+	if ( isLocked ) {
+		return null;
+	}
+
+	const itemsWithoutDefaultBlock = filter( items, ( item ) =>
+		item.name !== getDefaultBlockName() || ! isEmpty( item.initialAttributes )
```

Comment:
Why `! isEmpty( item.initialAttributes )` ?

Reply:
It's so that shared paragraph blocks don't get filtered out.

Reply:
As a casual observer, I'd have no hints from the context available to reach this conclusion. Even now, I'm not clear what it is about shared blocks that causes us to filter them.

I'd suggest some combination of code comment and/or [explaining variable](https://refactoring.com/catalog/extractVariable.html).